src/components/SeriesSearch/SeriesSearchForm.js

```diff
@@ -1,16 +1,58 @@
-import React, { useRef } from 'react';
+import React, { useState } from 'react';
+import { getTitleMatches } from '../../api/api';
+import { Link } from 'react-router-dom';
+import _ from 'lodash';
 
-export const SearchForm = ({ onSubmit }) => {
-  const inputRef = useRef();
-  const handleSubmit = (event) => {
+export const SeriesSearchForm = () => {
+  const [searchResults, setSearchResults] = useState([]);
+
+  const onHandleChange = _.debounce(async (text) => {
+    const {
+      data: { results },
+    } = await getTitleMatches(text);
+
+    results && results.length && setSearchResults(results);
```

Comment:
if the requests fail, results comes out to be undefined or empty object and it will overwrite the correct request